pkg/productinfo/alibaba/productinfo_alibaba.go

```diff
@@ -0,0 +1,207 @@
+package alibaba
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/log"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	"github.com/banzaicloud/productinfo/pkg/productinfo"
+	"sync"
+)
+
+// AlibabaInfoer encapsulates the data and operations needed to access external Alibaba resources
+type AlibabaInfoer struct {
+	client *ecs.Client
+}
+
+// NewAlibabaInfoer creates a new instance of the Alibaba infoer
+func NewAlibabaInfoer(regionId, accessKeyId, accessKeySecret string) (*AlibabaInfoer, error) {
+
+	// Create an ECS client
+	ecsClient, err := ecs.NewClientWithAccessKey(
+		regionId,
+		accessKeyId,
+		accessKeySecret,
+	)
+	if err != nil {
+		// Handle exceptions
+		panic(err)
+	}
+
+	return &AlibabaInfoer{
+		client: ecsClient,
+	}, nil
+}
+
+// Initialize downloads and parses the ECS price list on Alibaba Cloud
+func (e *AlibabaInfoer) Initialize() (map[string]map[string]productinfo.Price, error) {
+	log.Debug("initializing Alibaba price info")
+
+	var allPrices = struct {
+		sync.RWMutex
+		prices map[string]map[string]productinfo.Price
+	}{prices: make(map[string]map[string]productinfo.Price)}
+	var waitGroup sync.WaitGroup
+	zonesInRegions := make(map[string][]string)
+
+	regions, err := e.GetRegions()
+	if err != nil {
+		return nil, err
+	}
+
+	req := ecs.CreateDescribeInstanceTypesRequest()
+	req.RegionId = "eu-central-1"
+
+	vmSizes, err := e.client.DescribeInstanceTypes(req)
+	if err != nil {
+		return nil, err
+	}
+	instanceTypes := vmSizes.InstanceTypes.InstanceType
+	for region := range regions {
+		zones, err := e.GetZones(region)
+		if err != nil {
+			return nil, err
+		}
+		zonesInRegions[region] = zones
+		waitGroup.Add(1)
+		go func(region string, instanceTypes []ecs.InstanceType, zonesInRegions map[string][]string) {
```

Comment:
please try to collect prices per region and assemble the final result at the end - don't use locks (this may simplify the implementation and also be a bit faster)

please use the approach we discussed (write partial results in channel...)